channeldb/channel.go

```diff
@@ -3843,7 +3855,8 @@ func deserializeRevocationLogEntry(r io.Reader) (RevocationLog, error) {
 	var rl RevocationLog
 
 	err := ReadElements(r,
-		&rl.LocalBalance, &rl.RemoteBalance, &rl.FeePerKw, &rl.CommitTx,
+		&rl.LocalBalance, &rl.RemoteBalance, &rl.LocalOutputIndex,
+		&rl.RemoteOutputIndex, &rl.FeePerKw, &rl.CommitTx,
```

Comment:
If we want to be a bit more flexible here from the PoV of potentially making additional modifications to what we store (something something taproot?), then we can use a TLV serialization here, as it'll allow us to make modifications in a forwards and backwards compatible manner.

Reply:
Happened to check the TLV part for payment/invoice db optimization today, took me a while to understand how it did encoding, particularly the `BigSize` part that acted like a compression. The only downside I'd say is it introduces some overhead, but it does make things much more flexible. I think we can go for it. Just curious about what could be saved differently here?